internal/cmd/signer.go

```diff
@@ -14,27 +14,48 @@ import (
 func GetSigner(ctx *cli.Context) (notation.Signer, error) {
 	// read paths of the signing key and its corresponding cert.
 	var keyPath, certPath string
+	var pluginPath string
+	var kmsProfile config.KMSProfileSuite
+	var err error
 	if path := ctx.String(FlagKeyFile.Name); path != "" {
 		keyPath = path
 		certPath = ctx.String(FlagCertFile.Name)
 	} else {
-		var err error
 		keyPath, certPath, err = config.ResolveKeyPath(ctx.String(FlagKey.Name))
 		if err != nil {
-			return nil, err
+			if !errors.Is(err, config.ErrKeyNotFound) {
+				return nil, err
+			}
+
+			// check if the key is an external kms key
+			if kmsProfile, err = config.ResolveKMSKey(ctx.String(FlagKey.Name)); err != nil {
+				return nil, err
+			}
+			// get the plugin path for the external kms key
+			if pluginPath, err = config.ResolveKMSPluginPath(kmsProfile.PluginName); err != nil {
+				return nil, err
+			}
 		}
```

Comment:
The local key take precedence over the KMS key on the same key name.

Should we make the key name unique among all keys, including local keys and kms keys?